migen/build/generic_platform.py

```diff
@@ -1,4 +1,6 @@
 import os
+import shutil
+import re
```

Comment:
You are not using that import (and you probably want ``os.path`` facilities instead of your own regexps). Remove.